File: frontend/src/services/ticketService.js
import api from './api';

export const ticketService = {
  getAll: (params) => api.get('/tickets', { params }),
  getById: (id) => api.get(`/tickets/${id}`),
  create: (data) => api.post('/tickets', data),
  update: (id, data) => api.put(`/tickets/${id}`, data),
  updateStatus: (id, data) => api.patch(`/tickets/${id}/status`, data),
  assign: (id, data) => api.patch(`/tickets/${id}/assign`, data),
  rate: (id, rating) => api.patch(`/tickets/${id}/rate`, { rating }),
  reopen: (id) => api.patch(`/tickets/${id}/reopen`),
  updateResolutionNotes: (id, notes) => api.patch(`/tickets/${id}/resolution-notes`, { resolutionNotes: notes }),
  getResourceHistory: (resourceId, params) => api.get(`/tickets/resource/${resourceId}/history`, { params }),
  addAttachment: (id, file) => {
    const formData = new FormData();
    formData.append('file', file);
    return api.post(`/tickets/${id}/attachments`, formData, {
      headers: { 'Content-Type': 'multipart/form-data' },
    });
  },
  getAttachmentUrl: (ticketId, attachmentId) =>
    `/api/tickets/${ticketId}/attachments/${attachmentId}`,
  getComments: (id, params) => api.get(`/tickets/${id}/comments`, { params }),
  addComment: (id, data) => api.post(`/tickets/${id}/comments`, data),
  updateComment: (ticketId, commentId, data) =>
    api.put(`/tickets/${ticketId}/comments/${commentId}`, data),
  deleteComment: (ticketId, commentId) =>
    api.delete(`/tickets/${ticketId}/comments/${commentId}`),
};

File: backend/src/main/java/com/smartcampus/ticket/service/TicketService.java
package com.smartcampus.ticket.service;

import com.smartcampus.ticket.dto.*;
import com.smartcampus.ticket.entity.TicketPriority;
import com.smartcampus.ticket.entity.TicketStatus;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.web.multipart.MultipartFile;

public interface TicketService {

    TicketResponse create(TicketRequest request, Long userId);

    TicketResponse getById(Long id);

    Page<TicketResponse> getAll(TicketStatus status, TicketPriority priority, Long resourceId, Pageable pageable);

    Page<TicketResponse> getByUser(Long userId, Pageable pageable);

    TicketResponse update(Long id, TicketRequest request, Long userId);

    TicketResponse updateStatus(Long id, TicketStatusRequest request);

    TicketResponse assign(Long id, Long technicianId);

    AttachmentResponse addAttachment(Long ticketId, MultipartFile file, Long userId);

    byte[] getAttachmentFile(Long ticketId, Long attachmentId);

    String getAttachmentContentType(Long ticketId, Long attachmentId);

    Page<CommentResponse> getComments(Long ticketId, Pageable pageable);

    CommentResponse addComment(Long ticketId, CommentRequest request, Long userId);

    CommentResponse updateComment(Long ticketId, Long commentId, CommentRequest request, Long userId);

    void deleteComment(Long ticketId, Long commentId, Long userId);

    TicketResponse rate(Long ticketId, Integer rating, Long userId);

    TicketResponse updateResolutionNotes(Long ticketId, String notes);

    Page<TicketResponse> getResourceHistory(Long resourceId, Pageable pageable);

    TicketResponse reopen(Long ticketId, Long userId);
}
